alien4cloud-core/src/main/java/alien4cloud/deployment/DeploymentTopologyService.java

```diff
@@ -294,6 +308,10 @@ private void doUpdateDeploymentTopology(DeploymentTopology deploymentTopology, T
     public void updateDeploymentTopologyInputsAndSave(DeploymentTopology deploymentTopology) {
         deploymentInputService.processInputProperties(deploymentTopology);
         deploymentInputService.processProviderDeploymentProperties(deploymentTopology);
+        /// since inputs must be done before processing substitutions, we do it here
```

Comment:
good point. We could indeed check if a location has already been selected for the deploymentTopology before  injecting inputs and processing substitutions.